### PR TITLE
Hosted search

### PR DESCRIPTION
**Pull request by 1BE6-ankit**
API endpoints for search are added. The conan client can perform search in basically three ways:
1. Empty search - when nothing is specified in the search:
```
conan search -r conan-center
```
Then the following request will be made:
```
https://conan.bintray.com/v1/conans/search
```

2. Specifying some keywords: Requests for incomplete recipe names such as:
```
conan search Poco -r conan-center
conan search Poco/1.7.8p3@* -r conan-center
```
will have their requests as:
```
https://conan.bintray.com/v1/conans/search?=
```

3. When specifying full package reference:
```
conan search Poco/1.7.8p3@pocoproject/stable -r conan-center 
```
will have:
```
https://conan.bintray.com/v1/conans/Poco/1.7.8p3/pocoproject/stable/search?
```

Based on these requests necessary matchers, handlers and helpers were created.

This pull request makes the following changes:
* Matchers for requests in ConanHostedRecipe.groovy
* Handlers for the endpoints in HostedHandlers.java
* Helper package org.sonatype.repository.conan.internal.hosted.search to work on the search.
* Some minor changes after sonatype/codestyle was used to reformat the code

### CODE REVIEW

**Comment by DarthHater** [comment]
This is RAD. Thanks for submitting it! @bhamail @doddi or I will try and take a gander as soon as possible!

File: src/main/java/org/sonatype/repository/conan/internal/hosted/search/ConanHostedSearchFacetImpl.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.sonatype.nexus.repository.view.Context;
+
+@Named
+public class ConanHostedSearchFacetImpl
+    extends FacetSupport
+    implements ConanHostedSearchFacet
+{
+  @Override
+  public Content searchRecipes(Context context) throws IOException {
+    return null;
```
Comment (DarthHater) [inline src/main/java/org/sonatype/repository/conan/internal/hosted/search/ConanHostedSearchFacetImpl.java]: Errrm?

File: src/main/java/org/sonatype/repository/conan/internal/hosted/search/ConanHostedSearchFacetImpl.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  }
+
+  @Override
+  public Content searchPackages(Context context) throws IOException {
+    return null;
```
Comment (DarthHater) [inline src/main/java/org/sonatype/repository/conan/internal/hosted/search/ConanHostedSearchFacetImpl.java]: Here too :)

**Comment by DarthHater** [comment]
Took a quick look since we all were so excited, and noticed the implementation is missing :)

Anything we can do to help?

**Comment by 1BE6-ankit** [comment]
Sorry about that for not making it clear. New around the hood. With this pull request I was trying to have only the routes set up. I am also working on the functionality part. I thought I had search on recipes implemented, but on closer look there were more problems. Also, I thought for 'working on one thing at a time' contribution guideline, it would be better to have the routes set up first. With this I could also verify whether I have a proper structure of matchers, handlers, and helpers, and if this layout can be added to extend functionality by other contributors. Sorry for raising your expectations so high. Here on, should I try to implement all the search functionalities and have another request, or what would be the better way to work on this problem.
Thanks.

**Comment by DarthHater** [comment]
In this case, since it would be largely useless without the implementation, it's actually better to have it all in one PR. By work on one thing at a time, it would be like "implement one search route, all the way through, with tests", but your PR is far from gigantic. What we'd like to avoid is reviewing PRs that will take a few days to understand, you are in no danger of causing that at this point :)

**Comment by DarthHater** [comment]
PS, kudos for actually reading the guidelines, we appreciate it!!!!

**Comment by 1BE6-ankit** [comment]
Thanks for the review. I will try to have it proper in next PR.

**Pull request closed without merging**